model/textparse/openmetricsparse_test.go

```diff
@@ -89,7 +89,28 @@ baz_sum 324789.3
 baz_created 1520872607.123
 # HELP fizz_created Gauge which shouldn't be parsed as CT
 # TYPE fizz_created gauge
-fizz_created 17.0`
+fizz_created 17.0
```

Comment:
I would propose to split optimizations for test changes. Ideally don't want to accidently introduce functional bug because of some testing gap causes by testing change in the same PR 🤗  WDYT?

Reply:
Hmm, thing is the optimizations i made involves also fixing some stuff too.

because we're caching the CT in the parser, a lot of the cases in `TestCTParseFailures` fail (meaning theyre no longer failures)

Reply:
what about adding more test cases?
although ill have to check coverage to see if we've missed anything

Reply:
> because we're caching the CT in the parser, a lot of the cases in TestCTParseFailures fail (meaning theyre no longer failures)

Why? Are we not testing wrongly if the implementation changes the interface  functionality / semantics?

Reply:
> Why? Are we not testing wrongly if the implementation changes the interface functionality / semantics?

So before we'd have a CT parse failure if the created line was not the last line in a metric
Now it can be [above different metric lines](https://github.com/prometheus/prometheus/pull/14965/files#diff-9978684151c256395cd22e4883a7e22c14e153a567a769c8abed86a2225a1514R104) but _created cant be the [first line](https://github.com/prometheus/prometheus/pull/14965/files#diff-9978684151c256395cd22e4883a7e22c14e153a567a769c8abed86a2225a1514L796)

Reply:
for further clarification my understanding was `TestCTParseFailures` were failures that we want to fix down the line

Reply:
Just wanted to mention that I'm aligned with Manik here. This PR not only makes the parsing more efficient, it also fixes known limitations from the previous implementation.

One think you could do to make reviews easier @Maniktherana is to start organizing your commits, what do you think about:

* First commit with only test changes - which should trigger failures for `TestCTParseFailures` and `TestOpenMetricsParse`
* Second commit with optimization/fix - which should trigger successful tests